components/kserve/kserve.go

```diff
@@ -185,9 +186,17 @@ func (k *Kserve) configureServerless(instance *dsciv1.DSCInitializationSpec) err
 	case operatorv1.Managed: // standard workflow to create CR
 		switch instance.ServiceMesh.ManagementState {
 		case operatorv1.Unmanaged, operatorv1.Removed:
-			return fmt.Errorf("ServiceMesh is need to set to 'Managaed' in DSCI CR, it is required by KServe serving field")
+			return fmt.Errorf("ServiceMesh is need to set to 'Managed' in DSCI CR, it is required by KServe serving field")
 		}
-		serverlessInitializer := feature.NewFeaturesInitializer(instance, k.configureServerlessFeatures)
+		origin := featurev1.Origin{
+			Type: featurev1.ComponentType,
+			Name: k.GetComponentName(),
+		}
+		configureServerless := func(s *feature.FeaturesInitializer) error {
+			return k.configureServerlessFeatures(s, origin)
+		}
+
+		serverlessInitializer := feature.NewFeaturesInitializer(instance, configureServerless)
```

Comment:
How about aligning it to how it's done in `dashboard.go`?

That would mean:

```golang
		origin := featurev1.Origin{
			Type: featurev1.ComponentType,
			Name: k.GetComponentName(),
		}
		serverlessInitializer := feature.NewFeaturesInitializer(instance, k.configureServerlessFeatures(instance, origin))
```

and chaging `configureServerlessFeatures` to:

```golang
func (k *Kserve) configureServerlessFeatures(dscispec *dsci.DSCInitializationSpec, origin featurev1.Origin) feature.DefinedFeatures {
	return func(initializer *feature.FeaturesInitializer) error {
		// old code
	}
}
```